lib/src/util/number.dart

```diff
@@ -112,9 +112,139 @@ double fuzzyAssertRange(double number, int min, int max, [String? name]) {
 double moduloLikeSass(double num1, double num2) {
   if (num2 > 0) return num1 % num2;
   if (num2 == 0) return double.nan;
+  if (num2.isInfinite && ((num2 < 0 && num1 >= 0) || num2 >= 0 && num1 < 0)) {
+    return double.nan;
+  }
 
   // Dart has different mod-negative semantics than Ruby, and thus than
   // Sass.
   var result = num1 % num2;
   return result == 0 ? 0 : result + num2;
 }
+
+/// Returns the square root of [number].
+SassNumber sqrt(SassNumber number) {
+  number.assertNoUnits();
+  return SassNumber(math.sqrt(number.value));
+}
+
+/// Returns the sine of [number].
+SassNumber sin(SassNumber number) {
+  return SassNumber(math
+      .sin(number.assertNumber("number").coerceValueToUnit("rad", "number")));
+}
+
+/// Returns the cosine of [number].
+SassNumber cos(SassNumber number) {
+  return SassNumber(math
+      .cos(number.assertNumber("number").coerceValueToUnit("rad", "number")));
+}
+
+/// Returns tangent of [number].
+SassNumber tan(SassNumber number) {
+  return SassNumber(math
+      .tan(number.assertNumber("number").coerceValueToUnit("rad", "number")));
+}
+
+/// Returns the arctangent of [number].
+SassNumber atan(SassNumber number) {
+  return SassNumber.withUnits(math.atan(number.value) * 180 / math.pi,
+      numeratorUnits: ['deg']);
+}
+
+/// Return arcsin of [number]
+SassNumber asin(SassNumber number) {
+  return SassNumber.withUnits(math.asin(number.value) * 180 / math.pi,
+      numeratorUnits: ['deg']);
+}
+
+/// Return arccos of [number]
+SassNumber acos(SassNumber number) {
+  return SassNumber.withUnits(math.acos(number.value) * 180 / math.pi,
+      numeratorUnits: ['deg']);
+}
+
+/// Return the absolute of [number]
+SassNumber abs(SassNumber number) {
+  return SassNumber(number.value.abs());
+}
+
+/// Return log of [number]
+SassNumber log(SassNumber number, SassNumber? base) {
+  if (base != null) {
+    return SassNumber(math.log(number.value) / math.log(base.value));
+  }
+  return SassNumber(math.log(number.value));
+}
+
+/// Return [num1]^[num2]
+SassNumber pow(SassNumber num1, SassNumber num2) {
+  num1.assertNoUnits();
+  num2.assertNoUnits();
+  return SassNumber(math.pow(num1.value, num2.value));
+}
+
+/// Return atan2 for y and x.
+SassNumber atan2(SassNumber y, SassNumber x) {
+  return SassNumber.withUnits(
+      math.atan2(y.value, x.convertValueToMatch(y, 'x', 'y')) * 180 / math.pi,
+      numeratorUnits: ['deg']);
+}
+
+SassNumber negativeZero() {
+  return SassNumber((-0.5).truncateToDouble());
+}
+
+SassNumber roundWithStep(
+    SassString strategy, SassNumber number, SassNumber step) {
+  if (step.value == 0) return SassNumber(double.nan);
+  if (number.value.isInfinite) {
+    if (step.value.isInfinite) {
+      return SassNumber(double.nan);
```

Comment:
This and most of the other non-zero return values need to have units from one of the input values.

Reply:
👍